### PR TITLE
bank: added user password change

### LINKED ISSUE #53: bank: user password change
Users should be able to change their passwords for obvious security reasons.

### PR DESCRIPTION
**Pull request by gadjevb**
Added password change functionality for the users, now they can change their password at will. This will improve their security!

Fixes #53

### CODE REVIEW

File: frontend/app/app.js [hunk 1]
```diff
@@ -98,3 +101,33 @@ app.controller('LogoutCtrl', function ($scope, $http) {
...
+    if (newPassword === validateNewPassword) {
+
+      $scope.query = {};
+      $scope.query.oldPass = oldPassword;
+      $scope.query.newPass = newPassword;
+
+      $http.post("/v1/changePassword", $scope.query).then( function (response) {
+        $scope.message.success = "Password changed successfully!!!";
+        $scope.message.failure = "";
+      }, function (failure) {
```
Comment (gadjevb) [inline frontend/app/app.js]: I can't catch the error response, thus every time the user will see "Password changed successfully!!!" even if the change failed.

Comment (mgenov) [inline frontend/app/app.js]: You mean failure is not invoked or failure has no information about the failure?

Comment (gadjevb) [inline frontend/app/app.js]: Its not invoked.

Comment (mgenov) [inline frontend/app/app.js]: Then probably the interceptor is not dispatching it correctly. Try to comment the interceptor and check what is the behaviour. 

2 New Questions: 

1. What is returning server as response ? 
2. is the success callback invoked ?

Comment (gadjevb) [inline frontend/app/app.js]: On success the server is returning status 200, on error is returning status 400.

Comment (mgenov) [inline frontend/app/app.js]: What happens when you comment the interceptor ?

**Comment by gadjevb** [comment]
@mgenov now the interceptor will catch and return new promise of rejection.

File: frontend/app/app.js [hunk 2]
```diff
@@ -19,8 +22,12 @@ app.config(function ($routeProvider, $locationProvider, $httpProvider) {
 		return {
 			'responseError': function(rejection) {
 			  if (rejection.status == 401) {
-          window.location.href = "/login";
-        }
+                window.location.href = "/login";
+              }
+
+              if (rejection.status == 400) {
```
Comment (mgenov) [inline frontend/app/app.js]: Why only responses with status code 400 are rejected ?

File: src/main/java/com/clouway/bank/adapter/http/UserPasswordChangingService.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import com.google.sitebricks.http.Post;
+import org.springframework.security.crypto.bcrypt.BCrypt;
+
+import java.util.Optional;
+
+/**
+ * @author Borislav Gadjev <gadjevb@gmail.com>
+ */
+@Service
+@At("/v1/changePassword")
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/adapter/http/UserPasswordChangingService.java]: /v1/useraccount/changePassword

File: frontend/app/app.js (hunk 2 above)
Comment (gadjevb) [inline frontend/app/app.js]: I will make it for all!

**Comment by gadjevb** [comment]
@mgenov made the according changes!

File: frontend/app/app.test.js [hunk 4]
```diff
@@ -27,5 +27,6 @@ app.run(function($httpBackend) {
   $httpBackend.whenGET('/v1/transactions?isNext=false&startingFromCursor=').respond(transactions);
   $httpBackend.whenGET(regPath).passThrough();
   $httpBackend.whenGET('/v1/useraccount').respond({id: "123445", name: "John", balance: 23});
+  $httpBackend.whenPOST('/v1/changePassword').respond(200);
```
Comment (mgenov) [inline frontend/app/app.test.js]: .

**Pull request merged**